docs/processes/fuzzer_targets.md

```diff
@@ -0,0 +1,119 @@
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+#include "unicode/coll.h"
+#include "unicode/localpointer.h"
+#include "unicode/locid.h"
+#include "unicode/tblcoll.h"
+
+IcuEnvironment* env = new IcuEnvironment();
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  size_t unistr_size = size/2;
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[unistr_size]);
+  std::memcpy(fuzzbuff.get(), data, unistr_size * 2);
+  icu::UnicodeString fuzzstr(false, fuzzbuff.get(), unistr_size);
+
+  icu::LocalPointer<icu::RuleBasedCollator> col1(
+      new icu::RuleBasedCollator(fuzzstr, status));
+
+  return 0;
+}
+```
+
+The ICU API under test is the ```RuleBasedCollator(const UnicodeString &rules, UErrorCode &status)``` constructor. The code interprets the fuzzer data as UnicodeString and passes it to the constructor. And that is all. Specific error handling or return value verification is not required because the fuzzer will detect all memory issues by means of memory/address sanitizer findings.
+
+### Makefile.in changes
+
+ICU fuzzer targets are build and executed by the OSS-Fuzz project. On side of ICU they are compiled to assure that the code is syntactically correct and, as a sanity check, executed in the most basic manner, i.e. with minimal testdata and without ASAN or MSAN analysis.
+
+Add the new fuzzer target to the list of targets in the ```FUZZER_TARGETS``` variable in [```Makefile.in```](https://github.com/unicode-org/icu/blob/master/icu4c/source/test/fuzzer/Makefile.in). The new fuzzer target will then be built and executed as part of a normal ICU4C unit test run. Note that each fuzzer target becomes executable of its own. As such it is linked with the code in ```fuzzer_driver.cpp```, which contains the ```main()``` function.
+
+### Fuzzer seed corpus
+
+Any fuzzer seed data for a fuzzer target goes into a file with name ```<fuzzer_target>_seed_corpus.txt```. In many cases the input parameter of the ICU API under test is of type ```UnicodeString```, in case of which the seed data should be in UTF-16 format. As an example,see [collator_rulebased_fuzzer_seed_corpus.txt](https://github.com/unicode-org/icu/blob/master/icu4c/source/test/fuzzer/collator_rulebased_fuzzer_seed_corpus.txt).
+
+### Guidelines and tips
+
+*   Leave all randomness to the fuzzer. If a random selection of any kind is needed (e.g., of a locale), then use bytes from the fuzzer data to make the selection ([example](https://github.com/unicode-org/icu/blob/master/icu4c/source/test/fuzzer/break_iterator_fuzzer.cpp)).
+*   In many cases ICU unit tests can provide seed data or at least ideas for seed data. If the API under test requires a Unicode string then make sure that the seed data is in UTF-16 encoding. This can be achieved with e.g. the 'iconv' command or using an editor that saves test in UTF-16 command.
```

Comment:
what does "text in UTF-16 command" mean?

Reply:
That 'command' is a leftover word from a previous edit still left hanging.